### PR TITLE
Included Solution upgrade checkbox and logic in 'Deployment Settings' screen

### PR DESCRIPTION
**Pull request by RajeevPentyala**
This PR includes following tasks

- Solution upgrade option at Profile level.
- Pull Request Completion (Added a Toggle on Deployment Pane and also created a new 'Complete Pull Request' column in 'Deployment User Setting' table)
- Manage Solutions link fix for GCC regions.

Fixes microsoft/coe-starter-kit#639
Fixes microsoft/coe-starter-kit#3133
Fixes microsoft/coe-starter-kit#3011
Fixes microsoft/coe-starter-kit#2306

### CODE REVIEW

File: CenterofExcellenceALMAccelerator/SolutionPackage/CanvasApps/cat_poweropsdevopsedition_7f3f4_DocumentUri_msapp_src/Src/DeploymentSettingsScreen.fx.yaml [hunk 1]
```diff
@@ -3093,6 +3093,7 @@ DeploymentSettingsScreen As screen:
                         }
                     )
                 );
+                Collect(DeploymentSettings,{Name: "SolutionUpgrade",Value: If(chkUpgradeSolution.Value,"upgrade","update")});
```
Comment (mikefactorial) [inline CenterofExcellenceALMAccelerator/SolutionPackage/CanvasApps/cat_poweropsdevopsedition_7f3f4_DocumentUri_msapp_src/Src/DeploymentSettingsScreen.fx.yaml:3096]: A thought on refactoring this. If instead of setting "SolutionUpgrade" to "upgrade" or "update" you instead set "TriggerSolutionUpgrade" to true or false you could remove a lot of logic both in the app and in the pipelines where you are translating between the text "upgrade" and "update" to true or false and also not have two different variables "SolutionUpgrade" and "TriggerSolutionUpgrade" in the pipeline.

Comment (mikefactorial) [reply CenterofExcellenceALMAccelerator/SolutionPackage/CanvasApps/cat_poweropsdevopsedition_7f3f4_DocumentUri_msapp_src/Src/DeploymentSettingsScreen.fx.yaml:3096]: e.g. 

```
Collect(DeploymentSettings,{Name: "TriggerSolutionUpgrade",Value: chkUpgradeSolution.Value});
```

Also, since this is a Collect() have you tested what happens when you check / uncheck multiple times? Does it add duplicates to the collection?

File: CenterofExcellenceALMAccelerator/SolutionPackage/CanvasApps/cat_poweropsdevopsedition_7f3f4_DocumentUri_msapp_src/Src/DeploymentSettingsScreen.fx.yaml [hunk 2]
```diff
@@ -3513,6 +3532,13 @@ DeploymentSettingsScreen As screen:
                         UpdateContext({EnvironmentDeploymentSettings: ParseDeploymentSettings.Run("[]")}),
                         UpdateContext({EnvironmentDeploymentSettings: ParseDeploymentSettings.Run(locEnvironmentDeploymentSetting.Data)})
                     );
+                	// Read SolutionUpgrade tag
```
Comment (mikefactorial) [inline CenterofExcellenceALMAccelerator/SolutionPackage/CanvasApps/cat_poweropsdevopsedition_7f3f4_DocumentUri_msapp_src/Src/DeploymentSettingsScreen.fx.yaml:3535]: If refactored above this would be 

                	UpdateContext({solutionupgradetag: 
                			 LookUp(
                                        EnvironmentDeploymentSettings,
                                        Name = "TriggerSolutionUpgrade"
                                    ).Value
                			});

File: CenterofExcellenceALMAccelerator/SolutionPackage/CanvasApps/cat_poweropsdevopsedition_7f3f4_DocumentUri_msapp_src/Src/DeploymentSettingsScreen.fx.yaml [hunk 3]
```diff
@@ -3472,6 +3473,24 @@ DeploymentSettingsScreen As screen:
             Y: =204
             ZIndex: =43
 
+        chkUpgradeSolution As checkbox:
+            Default: =If(solutionupgradetag="upgrade",true,false)
```
Comment (mikefactorial) [inline CenterofExcellenceALMAccelerator/SolutionPackage/CanvasApps/cat_poweropsdevopsedition_7f3f4_DocumentUri_msapp_src/Src/DeploymentSettingsScreen.fx.yaml:3477]: If refactored as above this would be Default: = solutionupgradetag

File: CenterofExcellenceALMAccelerator/SolutionPackage/CanvasApps/cat_poweropsdevopsedition_7f3f4_DocumentUri_msapp_src/Src/DeploymentSettingsScreen.fx.yaml (hunk 1 above)
Comment (RajeevPentyala) [reply CenterofExcellenceALMAccelerator/SolutionPackage/CanvasApps/cat_poweropsdevopsedition_7f3f4_DocumentUri_msapp_src/Src/DeploymentSettingsScreen.fx.yaml]: Refractored the code by setting true/false instead of upgrade/update. However, chkUpgradeSolution.Value returns bool and can't set in json format so used Text(chkUpgradeSolution.Value)

File: CenterofExcellenceALMAccelerator/SolutionPackage/CanvasApps/cat_poweropsdevopsedition_7f3f4_DocumentUri_msapp_src/Src/DeploymentSettingsScreen.fx.yaml (hunk 2 above)
Comment (RajeevPentyala) [reply CenterofExcellenceALMAccelerator/SolutionPackage/CanvasApps/cat_poweropsdevopsedition_7f3f4_DocumentUri_msapp_src/Src/DeploymentSettingsScreen.fx.yaml:3536]: Refractored the code by setting true/false instead of upgrade/update.

File: CenterofExcellenceALMAccelerator/SolutionPackage/CanvasApps/cat_poweropsdevopsedition_7f3f4_DocumentUri_msapp_src/Src/DeploymentSettingsScreen.fx.yaml (hunk 3 above)
Comment (RajeevPentyala) [reply CenterofExcellenceALMAccelerator/SolutionPackage/CanvasApps/cat_poweropsdevopsedition_7f3f4_DocumentUri_msapp_src/Src/DeploymentSettingsScreen.fx.yaml]: Refractored the code by setting true/false instead of upgrade/update. However, true/false had to be set as text and used below condition to set the checkbox 'Default'.
If(solutionupgradetag="true",true,false)

**Comment by mikefactorial** [comment]
@RajeevPentyala for https://github.com/microsoft/coe-starter-kit/issues/2306 would like to add a Bypass Approval Default (note: I renamed the toggle in the app to Bypass Approval) similar to how we have a Bypass Prepare Default to give administrators the ability to not show the toggle but enforce all PRs to be auto approved.

image

**Review by mikefactorial — APPROVED** [review]
